src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs

```diff
@@ -1297,7 +1297,7 @@ internal unsafe bool AddChar(char ch, int numBytes)
                     {
                         // Throw maybe
                         _bytes -= numBytes;                                        // Didn't encode these bytes
-                        _enc.ThrowCharsOverflow(_decoder, _bytes <= _byteStart);    // Throw?
+                        _enc.ThrowCharsOverflow(_decoder, _chars == _charStart);    // Throw?
                         return false;                                           // No throw, but no store either
```

Comment:
We have the same pattern here: https://github.com/dotnet/runtime/blob/89bba37a92053f52e38d5a5c81a1f3510319dabf/src/libraries/System.Text.Encoding.CodePages/src/System/Text/EncodingCharBuffer.cs#L52-L56

Does it have the same problem - are we missing coverage for this path?

Reply:
Yes, we have the same issue here. There is no test coverage for this path, but, despite multiple attempts, I do not manage to make the test failing before applying the fix :/
These encodings are quite complex and unknown to me. I am not sure to be able to prove the fix is required but I would recommend to keep the `EncodingCharBuffers` implementations aligned.